pkg/envoy/lds/multicluster_gateway.go

```diff
@@ -0,0 +1,85 @@
+package lds
```

Comment:
Rename file to `gateway.go`, I'll be adding ingress gateway specific listeners here as we discussed offline.